File: dev/integration_tests/ios_host_app/Host/ViewController.m
// Copyright 2014 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#import "ViewController.h"
#import <Flutter/Flutter.h>
#import <FlutterPluginRegistrant/GeneratedPluginRegistrant.h>

@implementation ViewController

// Boiler-plate add-to-app demo. Not integration tested anywhere.
- (void)viewDidLoad {
    [super viewDidLoad];
    UIButton *button = [UIButton buttonWithType:UIButtonTypeCustom];
    [button addTarget:self
               action:@selector(handleButtonAction)
     forControlEvents:UIControlEventTouchUpInside];
    [button setTitle:@"Press me" forState:UIControlStateNormal];
    [button setBackgroundColor:[UIColor blueColor]];
    button.frame = CGRectMake(80.0, 210.0, 160.0, 40.0);
    [self.view addSubview:button];
}

- (void)handleButtonAction {
    FlutterViewController* flutterViewController = [[FlutterViewController alloc] init];
    [GeneratedPluginRegistrant registerWithRegistry:flutterViewController];
    [self presentViewController:flutterViewController animated:false completion:nil];
}

@end

File: dev/integration_tests/ios_host_app_swift/Host/ViewController.swift
// Copyright 2014 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import UIKit
import Flutter
import FlutterPluginRegistrant

class ViewController: UIViewController {
  var flutterEngine : FlutterEngine?;

  // Boiler-plate add-to-app demo. Not integration tested anywhere.
  override func viewDidLoad() {
    super.viewDidLoad()
    let button = UIButton(type:UIButton.ButtonType.custom)
    button.addTarget(self, action: #selector(handleButtonAction), for: .touchUpInside)
    button.setTitle("Press me", for: UIControl.State.normal)
    button.frame = CGRect(x: 80.0, y: 210.0, width: 160.0, height: 40.0)
    button.backgroundColor = UIColor.blue
    self.view.addSubview(button)
    self.flutterEngine = FlutterEngine(name: "io.flutter", project: nil);
  }

  @objc func handleButtonAction() {
    if let flutterEngine = flutterEngine as? FlutterEngine {
      GeneratedPluginRegistrant.register(with: flutterEngine);

      let flutterViewController = FlutterViewController(engine: flutterEngine, nibName: nil, bundle: nil);
      self.present(flutterViewController, animated: false, completion: nil)
    }
  }
}
